lib/metasploit/framework/login_scanner/redis.rb

```diff
@@ -14,6 +14,10 @@ class Redis
         include Metasploit::Framework::LoginScanner::RexSocket
         include Metasploit::Framework::Tcp::Client
 
+        # Required to be able to properly overwrite and invoke the original method
+        # since we do not reply on inheritance directly, but module inclusion.
+        alias parent_scan! scan!
```

Comment:
calling the `scan!` method from the included `Metasploit::Framework::LoginScanner::Base` module.

When I define the `scan!` method on the class (to change the behavior), I cannot use `super` to call the original one, as this results in an error, stating there's no parent class with such method, since we used a module.

Reply:
To clarify further, when attempting to use `super` in the `scan!` method, this is the error thrown:

```
NoMethodError: super: no superclass method `scan!' for #<Metasploit::Framework::LoginScanner::Redis
```

Reply:
Are you mixing up instance versus class methods? I think this should work in conjunction with `super`

```diff
-        def scan!(&block)
+        def self.scan!(&block)
```

Reply:
We use `include Metasploit::Framework::LoginScanner::Base`, this defines the `scan!` method as instance method. If we used `extend`, it would have been a class method.

Reply:
Oh it's the combo of `ActiveSupport::Concern` and `included` is causing the weirdness
Works for me :+1: